onnxruntime/core/optimizer/common_subexpression_elimination.cc

```diff
@@ -90,11 +90,12 @@ class EquivalenceClass {
 
   friend struct ::std::hash<EquivalenceClass>;
   friend std::vector<std::vector<const EquivalenceClass*>> Normalize(const Node& node, const std::vector<const EquivalenceClass*>& inputs);
-
-  explicit EquivalenceClass(const NodeArg* non_op_value)
+  
+  explicit EquivalenceClass(const NodeArg* non_op_value, const onnx::TensorProto* initializer_)
```

Comment:
Add a third constructor instead that accepts only the initializer. `non_op_value_` should then be renamed to `graph_input_` and set to nullptr in that constructor.